AzureFunctions/EventHubsToElastic/run.csx

```diff
@@ -0,0 +1,123 @@
+#r "System.Configuration"
+
+using System;
+using System.Configuration;
+using Elasticsearch.Net;
+using Nest;
+
+public static async Task Run(string myEventHubMessage, TraceWriter log)
+{
+    await SendEventToElasticSearch(myEventHubMessage, log);
+}
+
+private const string Setting_Es_Uri = "es_uri";
+private const string Setting_Es_Username = "es_username";
+private const string Setting_Es_Password = "es_password";
+private const string Setting_Es_Index_Name_Prefix = "es_index_name_prefix";
+private const string Error_Es_Uri_Missing = "Setting \"" + Setting_Es_Uri + "\" is missing.";
+private const string Error_Es_Username_Missing = "Setting \"" + Setting_Es_Username + "\" is missing.";
+private const string Error_Es_Password_Missing = "Setting \"" + Setting_Es_Password + "\" is missing.";
+private const string DefaultIndexNamePrefix = "wadeventhub";
+private const string EventDocumentTypeName = "event";
+
+private static async Task SendEventToElasticSearch(string myEventHubMessage, TraceWriter log)
+{
+    string esUri = ConfigurationManager.AppSettings[Setting_Es_Uri];
+    string username = ConfigurationManager.AppSettings[Setting_Es_Username];
+    string password = ConfigurationManager.AppSettings[Setting_Es_Password];
+    string indexNamePrefix = ConfigurationManager.AppSettings[Setting_Es_Index_Name_Prefix];
+
+    if (string.IsNullOrWhiteSpace(esUri))
+    {
+        ReportError(log, Error_Es_Uri_Missing, throwException: true);
+    }
+
+    if (string.IsNullOrWhiteSpace(username))
+    {
+        ReportError(log, Error_Es_Username_Missing, throwException: true);
+    }
+
+    if (string.IsNullOrWhiteSpace(password))
+    {
+        ReportError(log, Error_Es_Password_Missing, throwException: true);
+    }
+
+    if (string.IsNullOrWhiteSpace(indexNamePrefix))
+    {
+        log.Warning($"Setting \"{Setting_Es_Index_Name_Prefix}\" not specified, using default value \"{DefaultIndexNamePrefix}\".");
+        indexNamePrefix = DefaultIndexNamePrefix;
+    }
+
+    log.Info($"Sending Event Hub data to ElasticSearch at {esUri} ...");
+
+    ConnectionSettings connectionSettings = new ConnectionSettings(new Uri(esUri)).BasicAuthentication(username, password);
+    ElasticClient client = new ElasticClient(connectionSettings);
+
+    string indexName = GetIndexName(indexNamePrefix);
+    await EnsureIndexExists(indexName, client, log);
```

Comment:
We should probably cache the result of the EnsureIndexExists and check only once on start up and when the date has changed. It's a network call, so checking it for every message may be too much. After the index is created, the chance it fails is extremely low, so it's better to just check for an error from IndexAsync instead.